src/operation/intersection/RectangleIntersection.cpp

```diff
@@ -682,7 +691,12 @@ std::unique_ptr<geom::Geometry>
 RectangleIntersection::clip(const geom::Geometry& g, const Rectangle& rect)
 {
     RectangleIntersection ri(g, rect);
-    return ri.clip();
+    std::unique_ptr<geom::Geometry> result = ri.clip();
+
+    std::unique_ptr<ElevationModel> elevModel = ElevationModel::create(g);
```

Comment:
This can be conditioned on the input actually having Z values.

Reply:
done